solidity/contracts/libs/isms/MultisigIsmMetadata.sol

```diff
@@ -48,26 +50,43 @@ library MultisigIsmMetadata {
         pure
         returns (bytes32)
     {
-        return bytes32(_metadata[ORIGIN_MAILBOX_OFFSET:MERKLE_PROOF_OFFSET]);
+        return bytes32(_metadata[ORIGIN_MAILBOX_OFFSET:MESSAGE_ID_OFFSET]);
+    }
+
+    function messageId(bytes calldata _metadata)
```

Comment:
Should this go at the end? It's really only needed if we go the merkle proof route

Reply:
its also needed for signature recovery

Reply:
Sure but it can (and should) be derived from message in the non-proof case